src/GafferSceneUI/LightTool.cpp

```diff
@@ -445,6 +458,179 @@ IECoreGL::MeshPrimitivePtr cone( float height, float startRadius, float endRadiu
 	return result;
 }
 
+const float g_tipScale = 10.f;
+const float g_tipIconSize = 1.25f;
+const float g_tipIconOffset = -0.25f;
+const float g_tipIndent = 1.75f;
+const float g_tipLineSpacing = -1.5f;
+
+IECoreGL::MeshPrimitivePtr unitCone()
+{
+	static IECoreGL::MeshPrimitivePtr result;
+	if( result )
+	{
+		return result;
+	}
+
+	result = cone( 1.5f, 0.5f, 0 );
+	return result;
+}
+
+void drawSelectionTips(
+	const V3f &gadgetSpacePosition,
+	std::vector<const Inspector::Result *> inspections,
+	const std::string multiPlugDescription,
+	const std::string infoSuffix,
+	const Handle *handle,
+	const ViewportGadget *viewport,
+	const GafferUI::Style *style
+)
+{
+	std::vector<GraphComponent *> parameterSources;
+	std::vector<std::string> warningTips;
+	for( const auto &inspection : inspections )
+	{
+		if( auto source = inspection->source() )
+		{
+			EditScope *editScope = inspection->editScope();
+			if( !editScope || ( editScope && editScope->isAncestorOf( source ) ) )
+			{
+				parameterSources.push_back( source );
+			}
+			else
+			{
+				parameterSources.push_back( editScope );
+			}
+
+			if( inspection->editable() && !inspection->editWarning().empty() )
+			{
+				warningTips.push_back( inspection->editWarning() );
+			}
+			else if( !inspection->editable() )
+			{
+				warningTips.push_back( inspection->nonEditableReason() );
+			}
+		}
+	}
+
+	std::string parameterInfo;
+	if( parameterSources.size() == 1 )
+	{
+		parameterInfo = fmt::format(
+			"Editing : {}",
+			parameterSources.front()->relativeName( parameterSources.front()->ancestor<ScriptNode>() )
+		);
+	}
+	else if( parameterSources.size() > 1 )
+	{
+		Plug *commonAncestor = nullptr;
+		for( int i = 0, eI = parameterSources.size() - 1; i < eI; ++i )
+		{
+			Plug *a = parameterSources[i]->commonAncestor<Plug>( parameterSources[i + 1] );
+			if( !commonAncestor )
+			{
+				commonAncestor = a;
+			}
+			else if( a != commonAncestor )
+			{
+				commonAncestor = nullptr;
+				break;
+			}
```

Comment:
> Instead, could we find the common ancestor of all the plugs?

That's what I wanted to do originally, but thinking through it and some web searching made me think it might be a pretty time consuming thing to compute. 

> We seem to be only choosing a common ancestor if every pair of plugs has the same common ancestor as every other pair.

That's what I intended, as a poor-man's substitute for what I think is the expensive operation of finding the common ancestor of all plugs.

Maybe I'm overthinking it? I guess we won't be dealing with all that many plugs so even if the algorithm doesn't scale well it may not matter too much?

Would I do something similar to the implementation of `GraphComponent::commonAncestor()` but with a `std::vector<GraphComponent *> others` instead of a single `other`?

Reply:
I think something fairly naive like this would probably perform reasonably?

```
commonAncestor = plugs[0].commonAncestor( plugs[1] )
for plug in plugs[2:] :
    if commonAncestor.isAncestorOf( plug ) :
        continue
     else :
         commonAncestor = plug.commonAncestor( commonAncestor )
```

> Would I do something similar to the implementation of GraphComponent::commonAncestor() but with a std::vector<GraphComponent *> others instead of a single other?

I think `commonAncestor` is already a bit too esoteric to have been a method on GraphComponent anyway - it should really have been in GraphComponentAlgo (which we don't have yet). So I think your new function would be a free function like so : `GraphComponent *commonAncestor( vector<const GraphComponent *> &graphComponents )`. Or potentially it would be templated on an iterator range instead. I'd go for the `vector` for simplicity for now, and just keep it as a private function in `LightTool.cpp` with a todo about maybe templating and moving to GraphComponentAlgo at some point.